html/browsers/browsing-the-web/history-traversal/scroll-restoration-order.html

```diff
@@ -0,0 +1,74 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>History restoration order test</title>
+<meta name="assert" content="https://html.spec.whatwg.org/multipage/browsing-the-web.html#history-traversal">
+<meta name="assert" content="Traversing history should restore scroll position after dispatching popstate and before dispatching hashchange">
+
+<style>
+  body {
+    height: 200vh;
+    width: 200vw;
+  }
+</style>
+
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+  'use strict';
+  async_test(function(t) {
+    window.addEventListener('load', t.step_func(function() {
+      // Allow 1px epsilon for fractional scrolling.
+      assert_array_approx_equals(scrollPosition(), [0, 0], 1);
+
+      history.pushState('#1', '', '#1');
+      window.scrollTo(50, 100);
+      assert_array_approx_equals(scrollPosition(), [50, 100], 1);
+
+      history.pushState('#2', '', '#2');
+      window.scrollTo(100, 200);
+      assert_array_approx_equals(scrollPosition(), [100, 200], 1);
+
+      setTimeout(t.step_func(function(){
+        history.pushState(null, null, '#done');
+        window.scrollTo(555, 555);
+        assert_array_approx_equals(scrollPosition(), [555, 555], 1);
+        // Kick off the verification.
+        window.history.back();
+      }), 0);
+    }));
+
+    window.addEventListener('popstate', t.step_func(function() {
+      // Verify that scroll position is *not* restored before popstate.
+      const key = location.hash;
+      const expected_scroll_position = expectedScrollPositionForKey(key);
+      assert_not_equals(scrollPosition()[0], expected_scroll_position[0], `scroll is restored before popstate for ${key}`);
+      assert_not_equals(scrollPosition()[1], expected_scroll_position[1], `scroll is restored before popstate for ${key}`);
+
+      if (key == '')
```

Comment:
There's no JS code style lint, so nothing is enforced. https://web-platform-tests.org/writing-tests/general-guidelines.html doesn't say much. I think that if we had tooling that's able to automatically fixup things that doesn't pass the rules, then we could have more rules :)

Reply:
Lint isn't requirement for rules. Projects have had coding style rules forever, way before any automated tools. But ok, no rules.

Reply:
I know, I just think it'd be particularly hard to establish a common style in wpt without very strong tooling, both because all of the existing tests have random styles, and because the contributors come from and contribute via projects with different styles of their own.